torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -1156,14 +1157,18 @@ struct PythonPrintImpl {
         printOpName(stmt, node->kind());
         const FunctionSchema& schema = node->schema();
         stmt << "(";
-        for (size_t i = 0; i < node->inputs().size(); ++i) {
-          if (i > 0) {
+        // calculate how many args are specified.
+        // see (https://github.com/pytorch/pytorch/pull/56079) for more
+        // details.
+        size_t necessary_args =
+            CalculateNecessaryArgs(schema.arguments(), node->inputs());
+        for (size_t index = 0; index < necessary_args; ++index) {
+          if (index > 0)
             stmt << ", ";
-          }
-          auto v = useOf(node->inputs().at(i));
+          auto v = useOf(node->inputs().at(index));
           // print the kwarg name if it is a kwarg only argument.
-          if (i < schema.arguments().size()) {
-            auto arg = schema.arguments().at(i);
+          if (index < schema.arguments().size()) {
+            auto arg = schema.arguments().at(index);
```

Comment:
These changes are not necessary if we don't change the variable name from i to index

Reply:
done